filterBOM.js

```diff
@@ -4,6 +4,7 @@ function uniqueComponents(components) {
   const uniques = [];
   seen = {};
   for (const component of components) {
+    component.scope = "required";
```

Comment:
Just curious what does setting `scope` to `required` do here? A comment on why this is needed might be helpful. 

Also, this function looks to be destructive and mutates the input `components` arg. It's probs fine for this script but just a tad mean and unexpected. Might be nicer to avoid the mutation with something like this.

```js
seen[component.purl] = { ...component, scope: 'required' };
```

Reply:
Also something minor that doesn't need to be addressed in this PR: Since the unique values are being tracked by the `seen` object. The `uniques` array and iteration to populate it could be removed entirely by returning with something like `return Object.values(seen);`.

Reply:
Thanks, @charlieduong94! great feedback.